kernel/kernel-impl/src/main/java/org/sakaiproject/content/impl/ContentHostingComparator.java

```diff
@@ -102,28 +106,24 @@ public int compare(Object o1, Object o2)
 		{
 			Entity entity1 = (Entity) o1;
 			Entity entity2 = (Entity) o2;
-			String str1 = entity1.getProperties().getProperty(property);
-			String str2 = entity2.getProperties().getProperty(property);
+			Integer rank1 = null;
+			Integer rank2 = null;
 
-			if(str1 == null || str2 == null)
-			{
-				// ignore -- default to a different sort
-			}
-			else
-			{
-				try
-				{
-					Integer rank1 = new Integer(str1);
-					Integer rank2 = new Integer(str2);
-					return m_ascending ? rank1.compareTo(rank2) : rank2.compareTo(rank1) ;
-				}
-				catch(NumberFormatException e)
-				{
-					// ignore -- default to a different sort
-				}
-			}
-			// if unable to do a priority sort, sort by title
-			property = ResourceProperties.PROP_DISPLAY_NAME;
+			try {
+				rank1 = new Integer(entity1.getProperties().getProperty(property));
+			} catch (NumberFormatException e) { /* ignore, rank1 will be null */ }
+
+			try {
+				rank2 = new Integer(entity2.getProperties().getProperty(property));
+			} catch (NumberFormatException e) { /* ignore, rank2 will be null */ }
+
+			// Priorities can be null if a resource or collection was created before priorities were
+			// introduced. Sort null priorities to the bottom.
```

Comment:
Wouldn't it be better here to sort the remaining elements by displayname, rather than having them unsorted?